src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -67,4 +68,24 @@ def puppetVersion()
   def run_mode()
     Puppet.run_mode.name.to_s
   end
+
+  private
+
+  def params_to_ruby(params)
+    Hash[params.collect do |key, value|
+      # Values for query string and/or form parameters which are specified
+      # with array-like syntax will be parsed by Ring into a Clojure
+      # PersistentVector, which derives from a Java List.  Need to
+      # translate the Java List into a Ruby Array so that the request
+      # handling logic in Ruby can make use of it.
+
+      # For example, a query string of 'arr[]=one&arr[]=two" will be translated
```

Comment:
Presuming we don't pull in the `ring-nested-params` middleware, it would be better to have an example here that doesn't have square brackets in the parameter name.